token-metadata/program/src/utils/mod.rs

```diff
@@ -190,18 +192,42 @@ pub fn zero_account(s: &str, size: usize) -> String {
 pub fn close_program_account<'a>(
     account_info: &AccountInfo<'a>,
     funds_dest_account_info: &AccountInfo<'a>,
+    key: Key,
 ) -> ProgramResult {
+    let rent = Rent::get()?;
+
+    let rent_lamports = match key {
+        // Metadata accounts could have fees stored, so we only want to withdraw
+        // the actual rent lamport amount.
+        Key::MetadataV1 => rent.minimum_balance(Metadata::size()),
+        // Other accounts the rent is just the current lamport balance.
+        _ => account_info.lamports(),
+    };
+
+    let remaining_lamports = account_info
+        .lamports()
+        .checked_sub(rent_lamports)
+        .ok_or(MetadataError::NumericalOverflowError)?;
+
+    let redeem_lamports = account_info
+        .lamports()
+        .checked_sub(remaining_lamports)
+        .ok_or(MetadataError::NumericalOverflowError)?;
+
```

Comment:
This version of the code has been simplified since previously when you took into account the value of `Rent(size=0)`.    The logic you have now works as long as any fee we have is greater than `Rent(size=0)`.

I think `remaining_lamports` and `redeem_lamports` no longer need two separate calculations since the second one is not being used with regard to `Rent(size=0)`, so it is algebraically pretty much the same statement as the first calculation:
 ```
remaining_lamports = account_val - Rent(Metadata::size())
redeem_lamports = account_val - remaining_lamports

//substitute
redeem lamports = account_val - (account_val - Rent(Metadata::size()))

redeem lamports = Rent(Metadata::size())
```

I think all you need to with this version of the calculation is get the `remaining_lamports` (which enforces that `account_val - Rent(Metadata::size()) >= 0`).  And then you are just always redeeming the `rent_lamports` to the user.  (`rent_lamports + remaining_lamports = account_val`).

This also highlights the gap initially discussed if `fee < Rent(size=0)` since we are _always_ returning the rent amount without regard to what is left over.